### PR TITLE
Add OpenTelemetry distributed tracing example for Azure Managed DTS

### PR DESCRIPTION
**Pull request by torosent**
# Summary

## What changed?
- Added a new `distributed-tracing.ts` example in `examples/azure-managed/` that demonstrates end-to-end OpenTelemetry distributed tracing with the Durable Task JavaScript SDK
- Added a `docker-compose.yml` that starts the DTS Emulator (`mcr.microsoft.com/dts/emulator:latest`) and Jaeger for local trace visualization
- Added `.env.emulator` with pre-configured connection string for the local emulator
- Added a comprehensive `README.md` with step-by-step setup instructions for both emulator and cloud scenarios
- Renamed connection string env var from `AZURE_DTS_CONNECTION_STRING` to `DURABLE_TASK_SCHEDULER_CONNECTION_STRING` across examples and E2E tests
- Fixed type annotation issue with `Task<Promise<T>>` vs `Task<T>` in `index.ts` fan-out example

## Why is this change needed?
- Users need a working example showing how to enable and visualize distributed traces for durable orchestrations
- The SDK already supports OpenTelemetry tracing (via optional peer dependency), but there was no example demonstrating how to set it up with a trace collector
- The DTS Emulator Docker image enables local development without an Azure subscription

## Issues / work items
- Related: Distributed tracing support for JavaScript SDK

---

# Project checklist
- [x] Release notes are not required for the next release
- [x] Backport is not required
- [ ] All required tests have been added/updated (unit tests, E2E tests)
- [x] Breaking change?
  - No breaking changes. Only additions (new example files) and env var rename in examples/tests.

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot (VS Code agent mode)
- AI-assisted areas/files: `distributed-tracing.ts`, `docker-compose.yml`, `.env.emulator`, `README.md`
- What you changed after AI output: Reviewed and verified all OTel SDK usage, connection string patterns, and DTS emulator image reference

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Existing unit tests pass. This PR adds example code only (no new unit tests needed).

## Manual validation (only if runtime/behavior changed)
- Environment (OS, Node.js version, components): macOS, Node.js 22, Docker
- Steps + observed results:
  1. `docker compose up -d` — DTS Emulator + Jaeger started
  2. `cp .env.emulator .env` — configured for local emulator
  3. `npm run example -- ./examples/azure-managed/distributed-tracing.ts` — both orchestrations completed
  4. Opened Jaeger UI at http://localhost:16686 — traces visible under service `durabletask-js-tracing-example`

---

# Notes for reviewers
- The OTel packages (`@opentelemetry/sdk-node`, etc.) are **not** added to `package.json` — they are installed ad-hoc with `npm install --no-save` as documented in the README, keeping the repo dependency footprint small.
- The DTS Emulator image is `mcr.microsoft.com/dts/emulator:latest` with `Authentication=None` for local dev.

### CODE REVIEW

File: packages/durabletask-js/test/tracing.spec.ts
Comment (YunchuWang) [inline packages/durabletask-js/test/tracing.spec.ts]: sendevent tests are missing

File: packages/durabletask-js/test/tracing.spec.ts [hunk 1]
```diff
@@ -0,0 +1,690 @@
...
+    expect(parseTraceparent("00-123-456-01")).toBeUndefined();
+  });
+
+  it("should return undefined for traceparent with wrong number of parts", () => {
+    expect(parseTraceparent("00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331")).toBeUndefined();
+    expect(parseTraceparent("00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01-extra")).toBeUndefined();
+  });
+});
+
+describe("createTraceparent", () => {
```
Comment (YunchuWang) [inline packages/durabletask-js/test/tracing.spec.ts]: Missing error-path tests

Comment (torosent) [reply packages/durabletask-js/test/tracing.spec.ts]: fixed

File: packages/durabletask-js/src/worker/activity-executor.ts [hunk 2]
```diff
@@ -28,21 +29,32 @@ export class ActivityExecutor {
...
+      const encodedOutput = activityOutput ? JSON.stringify(activityOutput) : undefined;
 
-    return encodedOutput;
+      // Log activity completion (matching .NET EventId 604)
+      WorkerLogs.activityCompleted(this._logger, orchestrationId, name);
+
+      return encodedOutput;
+    } catch (e: unknown) {
+      const err = e instanceof Error ? e : new Error(String(e));
+      // Log activity failure (matching .NET EventId 605)
```
Comment (YunchuWang) [inline packages/durabletask-js/src/worker/activity-executor.ts]: ditto

File: packages/durabletask-js/src/utils/pb-helper.util.ts [hunk 3]
```diff
@@ -181,15 +181,30 @@ export function newSubOrchestrationFailedEvent(eventId: number, ex: Error): pb.H
...
+
+    let stack = "";
+    if (typeof e === "object" && e !== null && "stack" in e) {
+      const possibleStack = (e as { stack?: unknown }).stack;
+      stack = possibleStack != null ? String(possibleStack) : "";
+    }
+
+    const stringValueStackTrace = new StringValue();
+    stringValueStackTrace.setValue(stack);
+    failure.setStacktrace(stringValueStackTrace);
```
[the remark this answers is not in the document]
Comment (torosent) [reply packages/durabletask-js/src/utils/pb-helper.util.ts]: fixed

**Review by YunchuWang — APPROVED** [review]

File: packages/durabletask-js/src/worker/orchestration-executor.ts [hunk 4]
```diff
@@ -67,30 +70,33 @@ export class OrchestrationExecutor {
...
       const eventCount = Object.keys(ctx._pendingEvents).length;
-      this._logger.info(`${instanceId}: Waiting for ${taskCount} task(s) and ${eventCount} event(s) to complete...`);
+      WorkerLogs.orchestrationWaiting(this._logger, instanceId, taskCount, eventCount);
     } else if (
       ctx._completionStatus &&
       ctx._completionStatus !== pb.OrchestrationStatus.ORCHESTRATION_STATUS_CONTINUED_AS_NEW
     ) {
-      const completionStatusStr = getOrchestrationStatusStr(ctx._completionStatus);
-      this._logger.info(`${instanceId}: Orchestration completed with status ${completionStatusStr}`);
+      if (ctx._completionStatus === pb.OrchestrationStatus.ORCHESTRATION_STATUS_FAILED) {
```
[the remark this answers is not in the document]
Comment (torosent) [reply packages/durabletask-js/src/worker/orchestration-executor.ts]: fixed

**Review by YunchuWang — APPROVED** [review]
